core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2772,17 +2772,13 @@ private[spark] object Utils extends Logging {
     }
 
     val masterScheme = new URI(masterWithoutK8sPrefix).getScheme
-    val resolvedURL = masterScheme.toLowerCase(Locale.ROOT) match {
-      case "https" =>
+
+    val resolvedURL = Option(masterScheme).map(_.toLowerCase(Locale.ROOT)) match {
+      case Some("https") =>
         masterWithoutK8sPrefix
-      case "http" =>
+      case Some("http") =>
         logWarning("Kubernetes master URL uses HTTP instead of HTTPS.")
         masterWithoutK8sPrefix
-      case null =>
-        val resolvedURL = s"https://$masterWithoutK8sPrefix"
-        logInfo("No scheme specified for kubernetes master URL, so defaulting to https. Resolved " +
-          s"URL is $resolvedURL.")
-        resolvedURL
       case _ =>
```

Comment:
@yaooqinn . Although the current logic causes NPE, `getScheme` itself can return `null` according to the spec.
- https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#getScheme--

Reply:
In the new logic, `None` case should map to https://github.com/apache/spark/pull/27721/files#diff-d239aee594001f8391676e1047a0381eL2782-L2785 .

Reply:
if the scheme is undefined and the uri may be valid，then it will be handled by https://github.com/apache/spark/pull/27721/files#diff-d239aee594001f8391676e1047a0381eL2767-L2771 . So，the case None here means the uri is invalid

Reply:
for example `k8s:///https://host:port` will be resolved to `/https://host:port` first and if we map `None` to the old logic, the final uri will be `k8s://https:///https://host:port`

Reply:
Got it. The recovery logic seems to be not robust enough.